spring-cloud-stream-binders/spring-cloud-stream-binder-kafka/src/main/java/org/springframework/cloud/stream/binder/kafka/KafkaMessageChannelBinder.java

```diff
@@ -711,60 +586,22 @@ public void doManualAck(LinkedList<MessageHeaders> messageHeadersList) {
 		}
 	}
 
-	private class KafkaPropertiesAccessor extends DefaultBindingPropertiesAccessor {
-
-		public KafkaPropertiesAccessor(Properties properties) {
-			super(properties);
-		}
-
-		public int getNumberOfKafkaPartitionsForProducer() {
-			int nextModuleCount = getNextModuleCount();
-			if (nextModuleCount == 0) {
-				throw new IllegalArgumentException("Module count cannot be zero");
-			}
-			int nextModuleConcurrency = getProperty(BinderPropertyKeys.NEXT_MODULE_CONCURRENCY, defaultConcurrency);
-			int minKafkaPartitions = getMinPartitionCount(defaultMinPartitionCount);
-			return Math.max(minKafkaPartitions, nextModuleCount * nextModuleConcurrency);
-		}
-
-		public int getNumberOfKafkaPartitionsForConsumer() {
-			int concurrency = getConcurrency(defaultConcurrency);
-			int minKafkaPartitions = getMinPartitionCount(defaultMinPartitionCount);
-			int moduleCount = getCount();
-			if (moduleCount == 0) {
-				throw new IllegalArgumentException("Module count cannot be zero");
-			}
-			return Math.max(minKafkaPartitions, moduleCount * concurrency);
-		}
-
-		public String getCompressionCodec(String defaultValue) {
-			return getProperty(COMPRESSION_CODEC, defaultValue);
-		}
-
-		public int getRequiredAcks(int defaultRequiredAcks) {
-			return getProperty(REQUIRED_ACKS, defaultRequiredAcks);
-		}
-
-		public boolean getDefaultAutoCommitEnabled(boolean defaultAutoCommitEnabled) {
-			return getProperty(AUTO_COMMIT_ENABLED, defaultAutoCommitEnabled);
-		}
-
-		public int getMinPartitionCount(int defaultPartitionCount) {
-			return getProperty(BinderPropertyKeys.MIN_PARTITION_COUNT, defaultPartitionCount);
-		}
-
-	}
-
 	private class ReceivingHandler extends AbstractReplyProducingMessageHandler {
 
 		public ReceivingHandler() {
```

Comment:
I think this default constructor can be removed? and we still need to invoke `setBeanFactory` in the other constructor? not sure what's the significance of setting beanfactory here.